### PR TITLE
Use cookie-based auth for the Jupyter proxy

### PR DESCRIPTION
**Pull request by rtitle**
### Problem: 
Since the Leo proxy endpoint is under `/api/`, it requires a Google OAuth access token passed in the `Authorization` header _for every request_. However we can't (easily) hook into third party Jupyter javascript code to actually inject this header. 

### Solution: 
- Leo proxy listens under `/notebooks/`, not `/api/notebooks/`
- Leo proxy requires presence of a `FCToken` cookie (otherwise it returns a 401 status)
- The UI will need to set this cookie, similar to: https://github.com/broadinstitute/firecloud-ui/blob/ca7d3ce061c26d8053d4753d9057cd366d3a3a98/src/cljs/main/broadfcui/page/workspace/summary/attribute_editor.cljs#L116-L120

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by dvoet** [comment]
Where will the token actually be validated?

**Comment by rtitle** [comment]
My thinking was that (in the future) Leo will call Sam with the access token, and the sam call will do token validation (as well as resource validation for the notebook itself). Is that sufficient or does Leo need to do some validation of its own?

**Comment by rtitle** [comment]
Verified locally by creating a cluster, manually setting a cookie, and confirming I can access the notebook.

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala [hunk 1]
```diff
@@ -77,6 +77,16 @@ class LeoRoutes(val leonardoService: LeonardoService, val proxyService: ProxySer
     }
   }
 
+  private val rejectionHandler: RejectionHandler = {
+    RejectionHandler.newBuilder()
+      .handle {
+        case MissingCookieRejection(name) if name == tokenCookieName =>
+          complete(StatusCodes.Unauthorized,
+            ErrorReport("Unauthorized: Access is denied due to invalid credentials.", Some(StatusCodes.Unauthorized), Seq.empty, Seq.empty, Some(this.getClass)))
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala]: Exception class (last param) LeoRoutes?

File: jupyter-docker/jupyter_notebook_config.py [hunk 2]
```diff
@@ -14,5 +14,5 @@
 c.NotebookApp.allow_origin = '*'
 
 fragment = os.environ['GOOGLE_PROJECT'] + '/' + os.environ['CLUSTER_NAME']
-c.NotebookApp.base_url = '/api/notebooks/' + fragment + '/'
-c.NotebookApp.webapp_settings = {'static_url_prefix':'/api/notebooks/' + fragment + '/static/'}
\ No newline at end of file
+c.NotebookApp.base_url = '/notebooks/' + fragment + '/'
```
Comment (rtitle) [inline jupyter-docker/jupyter_notebook_config.py]: @ansingh7115 we'll need to rebuild the Jupyter docker image yet again. Is this process documented somewhere so I don't have to keep bugging you to do it?

(I might have yet another change for GAWB-2535 -- TBD)

Comment (rtitle) [reply jupyter-docker/jupyter_notebook_config.py]: Maybe a README.md under jupyter-docker/ would be helpful

Comment (jmthibault79) [reply jupyter-docker/jupyter_notebook_config.py]: Or a `build.sh`

**Pull request merged**